contractcourt/briefcase.go

```diff
@@ -1047,6 +1062,72 @@ func (b *boltArbitratorLog) FetchChainActions() (ChainActionMap, error) {
 	return actionsMap, nil
 }
 
+// LogLocalForceCloseInfo records the passed-in localForceCloseInfo
+// object within the ArbitratorLog for deferred consumption. This is
+// typically invoked when the channel force closure is first initiated
+// either by the user or automatically (link error, HTLC actions, etc.).
+func (b *boltArbitratorLog) LogLocalForceCloseInfo(
+	info channeldb.LocalForceCloseInfo) error {
+
+	return kvdb.Update(b.db, func(tx kvdb.RwTx) error {
+		scopeBucket, err := tx.CreateTopLevelBucket(b.scopeKey[:])
+		if err != nil {
+			return err
+		}
+		var buf bytes.Buffer
+		if err := channeldb.SerializeLocalCloseInfo(&buf, &info); err != nil {
+			return err
+		}
+
+		return scopeBucket.Put(localForceCloseInfoKey, buf.Bytes())
+	}, func() {})
+}
+
+// FetchLocalForceCloseInfo attempts to fetch the localForceCloseInfo
+// object that was previously logged in the ArbitratorLog. We use this
+// information when actually marking the channel as closed.
+func (b *boltArbitratorLog) FetchLocalForceCloseInfo() (
+	*channeldb.LocalForceCloseInfo,
+	error) {
+
+	var info *channeldb.LocalForceCloseInfo
+	infoExists := false
+	err := kvdb.View(b.db, func(tx kvdb.RTx) error {
```

Comment:
Could the passed in func return `(*channeldb.LocalForceCloseInfo, error)`? Then you could remove `infoExists` and further down do
```
if len(infoBytes) == 0 {
     return nil, fmt.Errorf("force close info does not exist...")
}
```

Reply:
Can I do that even though the function signature defined to be passed to `kvdb.View` is as defined here:
https://pkg.go.dev/github.com/lightningnetwork/lnd/kvdb@v1.4.4#View

Reply:
Apologies, my mistake, I thought you could decide on the `func` signature.